### PR TITLE
DOC-REFACTOR: 1/n Refactoring the documentation

### PR DESCRIPTION
**Pull request by aregm**
- added the first touch for developer docs.
- added component diagrams

Signed-off-by: Melik-Adamyan, Areg <areg.melik-adamyan@intel.com>

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -88,3 +98,5 @@
 html_sidebars = {
     "**": ["globaltoc.html", "relations.html", "sourcelink.html", "searchbox.html"]
 }
+
+issues_github_path = "modin-project/modin"
```
Comment (devin-petersohn) [inline docs/conf.py:102]: Missing newline at the end of the file

File: docs/developer/architecture.rst [hunk 2]
```diff
@@ -0,0 +1,260 @@
+System Architecture
```
Comment (devin-petersohn) [inline docs/developer/architecture.rst:1]: Do you want to delete the old version of the Architecture docs? Were there any changes from that?

File: docs/developer/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,164 @@
+Contributing
```
Comment (devin-petersohn) [inline docs/developer/contributing.rst:1]: Same questions as above on original contributing docs.

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
The commit message needs an open issue to pass the check. You can probably create one specific to this PR.

File: docs/developer/contributing.rst (hunk 3 above)
Comment (aregm) [reply docs/developer/contributing.rst:1]: Ditto.

File: docs/conf.py (hunk 1 above)
Comment (aregm) [reply docs/conf.py:102]: Fixed.

**Review by devin-petersohn — COMMENTED** [review]
Looks good!

**Pull request merged**